File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.py
def solution(n):
    answer = 0
    su = 0
    for i in range(n+1):
        start = i+1
        for j in range(start,n+1):
            su += j
            if su > n:
                su = 0
                break
            elif su == n:
                answer+=1
                su = 0
                break
    return answer

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n) {
    int answer = 0;
    // for(int i = 1; i <= n; i++){
    //     if(i%2==1){
    //         if((n/i)*i == n){
    //             answer += 1;
    //         }
    //     }
    //     else{
    //         if((n/i*2+1)*(i/2) == n){
    //             answer += 1;
    //         }
    //     }
    // }
    for(int i = 1; i <= n; i++){
        int sum = 0;
        for(int j = i; j <= n; j++){
            sum += j;
            if(sum == n){
                answer+=1;
                break;
            }
            else if(sum > n){
                break;
            }
        }
    }
    return answer;
}